src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,714 @@
+
+        // Check the liquidity provider caps after the new assets have been deposited
+        if (capManager != address(0)) {
+            ICapManager(capManager).postDepositHook(liquidityProvider, assets);
+        }
+
+        emit Deposit(liquidityProvider, assets, shares);
+    }
+
+    /// @notice Preview the amount of assets that would be received for burning a given amount of shares
+    /// @param shares The amount of shares to burn
+    /// @return assets The amount of liquidity assets that would be received
+    function previewRedeem(uint256 shares) external view returns (uint256 assets) {
+        assets = convertToAssets(shares);
+    }
+
+    /// @notice Request to redeem liquidity provider shares for liquidity assets
+    /// @param shares The amount of shares the redeemer wants to burn for liquidity assets
+    /// @return requestId The index of the withdrawal request
+    /// @return assets The amount of liquidity assets that will be claimable by the redeemer
+    function requestRedeem(uint256 shares) external returns (uint256 requestId, uint256 assets) {
+        // Calculate the amount of assets to transfer to the redeemer
+        assets = convertToAssets(shares);
+
+        requestId = nextWithdrawalIndex;
+        uint120 queued = SafeCast.toUint120(withdrawsQueued + assets);
+        uint40 claimTimestamp = uint40(block.timestamp + claimDelay);
+
+        // Store the next withdrawal request
+        nextWithdrawalIndex = SafeCast.toUint16(requestId + 1);
+        // Store the updated queued amount which reserves liquidity assets (WETH) in the withdrawal queue
+        withdrawsQueued = queued;
+        // Store requests
+        withdrawalRequests[requestId] = WithdrawalRequest({
+            withdrawer: msg.sender,
+            claimed: false,
+            claimTimestamp: claimTimestamp,
+            assets: SafeCast.toUint120(assets),
+            queued: queued
+        });
+
+        // burn redeemer's shares
+        _burn(msg.sender, shares);
+
+        // Remove the redeemed assets from the last available assets
+        lastAvailableAssets -= SafeCast.toInt128(SafeCast.toInt256(assets));
+
+        emit RedeemRequested(msg.sender, requestId, assets, queued, claimTimestamp);
+    }
+
+    /// @notice Claim liquidity assets from a previous withdrawal request after the claim delay has passed
+    /// @param requestId The index of the withdrawal request
+    /// @return assets The amount of liquidity assets that were transferred to the redeemer
+    function claimRedeem(uint256 requestId) external returns (uint256 assets) {
+        // Load the structs from storage into memory
+        WithdrawalRequest memory request = withdrawalRequests[requestId];
+
+        require(request.claimTimestamp <= block.timestamp, "Claim delay not met");
+        // Is there enough liquidity to claim this request?
+        require(
+            request.queued <= withdrawsClaimed + IERC20(liquidityAsset).balanceOf(address(this)),
```

Comment:
⚪️ Cleaner if we dedupe code and do `request.queued <= claimable()` here. `claimable()` becomes public.

Reply:
Fix proposed in this commit: [4990395](https://github.com/OriginProtocol/arm-oeth/pull/33/commits/499039528b97ab2a586846d7bfe119ec8fc76e2c).